src/create-template-orders.js

```diff
@@ -20,20 +33,263 @@ async function _fetchLastStartTimestamp() {
+      `Failed to create template order from the checkout order with number ${checkoutOrder.orderNumber}. ` +
+        'Skipping this checkout order' +
+        ` Error: ${JSON.stringify(serializeError(err))}`
+    )
+  }
+}
+
+function isDuplicateOrderError(e) {
+  return (
+    e.code === 400 &&
+    e.body?.errors?.length === 1 &&
+    e.body?.errors?.some(
+      (err) => err.code === 'DuplicateField' && err.field === 'orderNumber'
+    )
+  )
+}
+
+function _generateTemplateOrderDrafts(checkoutOrder) {
+  const orderDrafts = []
+  const subscriptionLineItems = checkoutOrder.lineItems.filter(
+    (lineItem) => lineItem.custom?.fields?.isSubscription
+  )
+  for (const lineItem of subscriptionLineItems)
+    if (lineItem.quantity > 1)
+      for (let i = 0; i < lineItem.quantity; i++) {
+        const templateOrderDraft = _generateTemplateOrderImportDraft(
+          checkoutOrder,
+          lineItem,
+          i + 1
+        )
+        orderDrafts.push(templateOrderDraft)
+      }
+    else {
+      const templateOrderDraft = _generateTemplateOrderImportDraft(
+        checkoutOrder,
+        lineItem
+      )
+      orderDrafts.push(templateOrderDraft)
+    }
+
+  return orderDrafts
+}
+
+function _generateTemplateOrderImportDraft(
+  checkoutOrder,
+  lineItem,
+  quantityIncrement
+) {
+  const schedule = lineItem.custom.fields.schedule
+  const cronExpression = parser.parseExpression(schedule)
+  let nextDeliveryCronDate = cronExpression.next()
+  const nextDeliveryDate = nextDeliveryCronDate.toDate()
+  let nextDeliveryDateISOString = nextDeliveryDate.toISOString()
+  const cutoffDays = lineItem.custom.fields.cutoffDays
+  if (cutoffDays) {
+    nextDeliveryDate.setDate(nextDeliveryDate.getDate() - cutoffDays)
+    if (nextDeliveryDate.toISOString() < checkoutOrder.createdAt) {
+      nextDeliveryDateISOString = nextDeliveryDate.toISOString()
+      nextDeliveryCronDate = cronExpression.next()
+    }
+  }
+  const reminderDays = lineItem.custom.fields.reminderDays
+  let nextReminderDateISOString
+  if (reminderDays) {
+    nextDeliveryCronDate.setDate(nextDeliveryCronDate.getDate() - reminderDays)
+    nextReminderDateISOString = nextDeliveryCronDate.toISOString()
+  }
+  const templateOrder = {
+    orderNumber:
+      lineItem.custom.fields.subscriptionKey +
+      (quantityIncrement ? `-${quantityIncrement}` : ''),
+    customerEmail: checkoutOrder.customerEmail,
+    totalPrice: lineItem.totalPrice,
+    lineItems: [lineItem],
+    custom: {
+      type: {
+        typeId: 'type',
+        key: 'subscription-template-order',
+      },
+      fields: {
+        ...checkoutOrder.custom.fields,
+        nextDeliveryDate: nextDeliveryDateISOString,
+        schedule,
+        checkoutOrderRef: { typeId: 'order', id: checkoutOrder.id },
+        reminderDays,
+        nextReminderDate: nextReminderDateISOString,
+      },
+    },
+    inventoryMode: 'None',
+    state: {
+      typeId: 'state',
+      key: 'Active',
+    },
+  }
```

Comment:
I think we should copy all properties/fields if it's set on the checkout order ? wdyt ?

Reply:
I copied the custom fields: https://github.com/commercetools/commercetools-subscriptions/pull/5/files#diff-33e44438e7461ee9bc6d2f6720f5f57e1aea2806100eb8ca03d98eb8b2124fc5R232. The other fields I am not sure because the checkout order could also contain other line items and fields like price, discounts etc might not be relevant for this.

Reply:
I added some more fields: https://github.com/commercetools/commercetools-subscriptions/pull/5/commits/da4bc688ef50224b6c989e7cf234b68d680fe8f2, I will put it this question into our slack channel.